source/extensions/common/wasm/wasm.cc

```diff
@@ -1474,8 +1475,9 @@ bool Wasm::initialize(const std::string& code, absl::string_view name, bool allo
   general_context_ = createContext();
   wasm_vm_->start(general_context_.get());
   if (is_emscripten_) {
-    ASSERT(std::isnan(emscripten_NaN_->get()));
-    ASSERT(std::isinf(emscripten_Infinity_->get()));
+    // TODO(PiotrSikora): re-enable those checks after v8 starts returning globals.
+    // ASSERT(std::isnan(emscripten_NaN_->get()));
```

Comment:
Didn't you fix these?

Reply:
I did & I didn't... This is something I forgot to discuss with you yesterday.

The way globals are implemented in V8, the VM retains ownership of the module (same as functions), but the current model passes ownership back to the abstraction layer, where it's never used (other than for those 2 asserts). Do we need to do that?

Reply:
There should be a way to get the value from v8 for a global so that the Global object is just a proxy for calls into v8?

Reply:
I could add a proxy, yes... My point is, that's an extra code that doesn't provide any value, since we never access globals from the host (outside of those 2 asserts).

Reply:
Depending on how and whether we end up supporting threads in the context of emscripten we may need to modify the internal emscripten value stack variables.  So, while we are not using the globals right now it doesn't mean we will not use them in the future.  That said, we don't need the support right now and I'd rather get v8 in, so let's leave a TODO and add it to the spreadsheet and resolve this conversation :)

Reply:
Added proxy, let me know if this works for you.